### DISCUSSION TITLE
Bitmap conversion changes rendered values beyond clipping. Why?

### DISCUSSION THREAD

**Discussion by AZaitzeff** [mitsuba-renderer/mitsuba3#909] [Q&A] [answered]
```
import drjit as dr
import mitsuba as mi
import matplotlib.pyplot as plt
import numpy as np
mi.set_variant("scalar_rgb")
scene = mi.load_file('../scenes/cbox.xml')
image_ref = mi.render(scene, spp=1024)
plt.imshow(image_ref)
```
![Screenshot 2023-09-18 at 4 49 37 PM](https://github.com/mitsuba-renderer/mitsuba3/assets/2396638/aeb3c6b7-b686-4c78-9c77-fa000ff606cf)

```
bitmap=mi.util.convert_to_bitmap(image_ref)
bitmap
```
![Screenshot 2023-09-18 at 4 49 31 PM](https://github.com/mitsuba-renderer/mitsuba3/assets/2396638/f67c7b93-b4f1-4a71-868d-17367b7a53e0)
```
bit_values=mi.TensorXf(bitmap).numpy().flatten()
render_values=image_ref.numpy().flatten()
plt.scatter(render_values,bit_values)
plt.xlim(0,1)
plt.ylim(0,255)

```
![Screenshot 2023-09-18 at 4 49 56 PM](https://github.com/mitsuba-renderer/mitsuba3/assets/2396638/f9620eae-17f6-4c00-b8b2-1fad6ef6abd0)
```
plt.imshow(np.sqrt(np.clip(image_ref,0,1)))
```
![Screenshot 2023-09-18 at 4 49 44 PM](https://github.com/mitsuba-renderer/mitsuba3/assets/2396638/be066cbf-2d91-4020-89d5-37015ebfbb70)

So it seems that in addition to clipping, `convert_to_bitmap` takes the square root (or something close to it) of the RGB values. Why does convert to bitmap change the RGB values in this way and is this documented somewhere?

**Comment by njroussel** [comment, marked as answer]
Hi @AZaitzeff 

Here's the source code of `convert_to_bitmap`: [source](https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/python/python/util.py#L527-L544). The documentation is also in our API reference.

I believe what you are referencing here is the sRGB gamma ramp/correction. We don't have a page about this topic specifically, but we do have pointers on how to enable/disable it in Mitsuba [this guide](https://mitsuba.readthedocs.io/en/stable/src/how_to_guides/image_io_and_manipulation.html).
In short, we need a linear color space to during rendering but most likely you'll want a sRGB color values when displaying the rendered image.

Also `convert_to_bitmap` is a convenience function. As you can see from its source code, it's only a couple of lines. You can use the full `BItmap` constructor and its methods if you wish to produce a different result.
